### ISSUE TITLE
Configuration file

### ISSUE THREAD

**Issue by rgarrigue** [koalaman/shellcheck#2299]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Question / Feature request

I saw no mention of this, so to be sure is there a way to configure shellcheck through a file like `.shellcheck` ?

If there's none, this is a feature request :-) And I'ld like to suggest ansible-lint as a great example. The config file is a `.ansible-lint` yaml formatted file like this one

```yaml
skip_list:
  - role-name  # Role name {0} does not match ``^[a-z][a-z0-9_]+$`` pattern

warn_list:
  - empty-string-compare  # Don't compare to empty string
  - no-changed-when  # Commands should not change things if nothing needs doing
  - no-handler  # Tasks that run when changed should likely be handlers

exclude_paths:
  - '../../.ansible'
```

And running ansible-lint tells you how to write .ansible-lint to ignore the issues if you feel like it

![image](https://user-images.githubusercontent.com/1872031/129338714-55605a5d-bf5a-4484-8493-5ce3c935cc4f.png)

**Comment by koalaman** [comment]
ShellCheck supports the use of a `.shellcheckrc`. Here's the documentation as it currently exists [in the man page](https://github.com/koalaman/shellcheck/blob/master/shellcheck.1.md):

## RC FILES                                                                                                                                                                                                       
                                                                                                                                                                                                                 
Unless `--norc` is used, ShellCheck will look for a file `.shellcheckrc` or                                                                                                                                      
`shellcheckrc` in the script's directory and each parent directory. If found,                                                                                                                                    
it will read `key=value` pairs from it and treat them as file-wide directives.                                                                                                                                   
                                                                                                                                                                                                                 
Here is an example `.shellcheckrc`:                                                                                                                                                                              
                                                                                                                                                                                                                 
    # Look for 'source'd files relative to the checked script,                                                                                                                                                   
    # and also look for absolute paths in /mnt/chroot                                                                                                                                                            
    source-path=SCRIPTDIR                                                                                                                                                                                        
    source-path=/mnt/chroot                                                                                                                                                                                      
                                                                                                                                                                                                                 
    # Turn on warnings for unquoted variables with safe values                                                                                                                                                   
    enable=quote-safe-variables                                                                                                                                                                                  
                                                                                                                                                                                                                 
    # Turn on warnings for unassigned uppercase variables                                                                                                                                                        
    enable=check-unassigned-uppercase                                                                                                                                                                            
                                                                                                                                                                                                                 
    # Allow [ ! -z foo ] instead of suggesting -n                                                                                                                                                                
    disable=SC2236                                                                                                                                                                                               
                                                                                                                                                                                                                 
If no `.shellcheckrc` is found in any of the parent directories, ShellCheck                                                                                                                                      
will look in `~/.shellcheckrc` followed by the XDG config directory                                                                                                                                              
(usually `~/.config/shellcheckrc`) on Unix, or `%APPDATA%/shellcheckrc` on                                                                                                                                       
Windows. Only the first file found will be used.                                                                                                                                                                 
                                                                                                                                                                                                                 
Note for Snap users: the Snap sandbox disallows access to hidden files.                                                                                                                                          
Use `shellcheckrc` without the dot instead.                                                                                                                                                                      
                                                                                                                                                                                                                 
Note for Docker users: ShellCheck will only be able to look for files that                                                                                                                                       
are mounted in the container, so `~/.shellcheckrc` will not be read.
